tests/Filesystem/FilesystemTest.php

```diff
@@ -137,32 +137,163 @@ public function testCopyDirectoryMovesEntireDirectory()
+		@unlink(__DIR__.'/file.txt');
+	}
+
+
+	public function testMoveMovesFiles()
+	{
+		file_put_contents(__DIR__.'/foo.txt', 'foo');
+		$files = new Filesystem;
+		$files->move(__DIR__.'/foo.txt',__DIR__.'/bar.txt');
+		$this->assertFileExists(__DIR__.'/bar.txt');
+		$this->assertFileNotExists(__DIR__.'/foo.txt');
+		@unlink(__DIR__.'/bar.txt');
+	}
+
+
+	public function testExtensionReturnsExtension()
+	{
+		file_put_contents(__DIR__.'/foo.txt', 'foo');
+		$files = new Filesystem;
+		$this->assertEquals('txt',$files->extension(__DIR__.'/foo.txt'));
+		@unlink(__DIR__.'/foo.txt');
+	}
+
+
+	public function testTypeIndentifiesFile()
+	{
+		file_put_contents(__DIR__.'/foo.txt', 'foo');
+		$files = new Filesystem;
+		$this->assertEquals('file',$files->type(__DIR__.'/foo.txt'));
+		@unlink(__DIR__.'/foo.txt');
+	}
+
+
+	public function testTypeIndentifiesDirectory()
+	{
+		mkdir(__DIR__.'/foo');
+		$files = new Filesystem;
+		$this->assertEquals('dir',$files->type(__DIR__.'/foo'));
+		@rmdir(__DIR__.'/foo');
+	}
+
+
+	public function testSizeOutputsSize()
+	{
+		$size = file_put_contents(__DIR__.'/foo.txt', 'foo');
+		$files = new Filesystem;
+		$this->assertEquals($size,$files->size(__DIR__.'/foo.txt'));
+		@unlink(__DIR__.'/foo.txt');
 	}
 
+
+	public function testLastModified()
+	{
+		$time = time();
+		file_put_contents(__DIR__.'/foo.txt', 'foo');
+		$files = new Filesystem;
+		$this->assertEquals($time,$files->lastModified(__DIR__.'/foo.txt'));
+		@unlink(__DIR__.'/foo.txt');
+	}
+
+
+	public function testIsWritable()
+	{
+		file_put_contents(__DIR__.'/foo.txt', 'foo');
+		$files = new Filesystem;
+		@chmod(__DIR__.'/foo.txt', 0444);
+		$this->assertFalse($files->isWritable(__DIR__.'/foo.txt'));
+		@chmod(__DIR__.'/foo.txt', 0777);
+		$this->assertTrue($files->isWritable(__DIR__.'/foo.txt'));
+		@unlink(__DIR__.'/foo.txt');
+	}
+
+
+	public function testGlobFindsFiles()
+	{
+		file_put_contents(__DIR__.'/foo.txt', 'foo');
+		file_put_contents(__DIR__.'/bar.txt', 'bar');
+		$files = new Filesystem;
+		$glob = $files->glob(__DIR__.'/*.txt');
+		$this->assertContains(__DIR__.'/foo.txt',$glob);
+		$this->assertContains(__DIR__.'/bar.txt',$glob);
+		@unlink(__DIR__.'/foo.txt');
+		@unlink(__DIR__.'/bar.txt');
+	}
+
+
+	public function testAllFilesFindsFiles()
+	{
+		file_put_contents(__DIR__.'/foo.txt', 'foo');
+		file_put_contents(__DIR__.'/bar.txt', 'bar');
+		$files = new Filesystem;
+		$allFiles = [];
+		foreach($files->allFiles(__DIR__) as $file)
+			$allFiles[] = $file->getFilename();
+		$this->assertContains('foo.txt',$allFiles);
+		$this->assertContains('bar.txt',$allFiles);
+		@unlink(__DIR__.'/foo.txt');
+		@unlink(__DIR__.'/bar.txt');
+	}
+
+
+	public function testDirectoriesFindsDirectories()
+	{
+		mkdir(__DIR__.'/foo');
+		mkdir(__DIR__.'/bar');
+		$files = new Filesystem;
+		$directories = $files->directories(__DIR__);
+		$this->assertContains(__DIR__.DIRECTORY_SEPARATOR.'foo',$directories);
```

Comment:
@lucasmichot When we get paths from the system, for example by `glob()`, the directory separator used is according to the system, for Windows, it's a `\`, so it's best to use `DIRECTORY_SEPARATOR`, when we are equating the paths as strings.